libmultilabel/linear/preprocessor.py

```diff
@@ -177,14 +177,18 @@ def as_ints(str):
             nz = 0
             for e in features.split():
                 ind, val = e.split(':')
-                val = float(val)
+                ind, val = int(ind), float(val)
+                if ind < 1:
+                    raise IndexError(f'invalid svm format at line {i+1} of the file \'{file_path}\' --> Indices should start from one.')
                 if val != 0:
-                    col_idx.append(int(ind) - 1)
+                    col_idx.append(ind - 1)
                     prob_x.append(val)
                     nz += 1
             row_ptr.append(row_ptr[-1]+nz)
+        except IndexError:
+            raise
         except:
-            raise ValueError(f'invalid svm format at line {i+1}')
+            raise ValueError(f'invalid svm format at line {i+1} of the file \'{file_path}\'') from None
```

Comment:
in our discussion we conclude that "from None" shouldn't be there as we want the error to be shown?

Reply:
I fixed it.